.github/workflows/integration_test_run.yaml

```diff
@@ -234,6 +234,14 @@ jobs:
       - name: Pre-run script
         if: ${{ inputs.pre-run-script != '' }}
         run: bash -xe ${{ inputs.pre-run-script }}
+      - name: Download charm artifact
+        uses: actions/download-artifact@v4
+        if: ${{ github.event_name == 'pull_request' }}
```

Comment:
I think you just moved it here, but maybe you can remove that line? I think we can also run operator workflows on other events like `push`.

Reply:
If the artifacts are built on a push event, it should be able to download them with this pattern. If they aren't there, no pattern will match the download-artifact step should continue without failure.

Reply:
How can they be downloaded if the step is not executed with `github.event.name` set to `push`?

I have seen that you have updated the code, so I am resolving this conversation.